helios-testing/src/main/java/com/spotify/helios/testing/SoloAgentProber.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright (c) 2014 Spotify AB.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.helios.testing;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+import com.spotify.helios.client.HeliosClient;
+import com.spotify.helios.common.descriptors.HostStatus;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+/**
+ * This class is used by {@link HeliosSoloDeployment} to check whether there are agents
+ * available in helios-solo.
+ */
+class SoloAgentProber {
+
+  private static final Logger log = LoggerFactory.getLogger(SoloAgentProber.class);
+
+  /**
+   * @param client {@link HeliosClient}
+   * @return true if we can connect to the specified host and port. Null, otherwise.
```

Comment:
I see that this signature is somewhat forced on this class by `Polling.awaitUnchecked`, but a method that can only ever return `true` or `null` feels pretty weird